spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/KubernetesCatalogWatch.java

```diff
@@ -113,4 +82,34 @@ public void catalogServicesWatch() {
 		}
 	}
 
+	@PostConstruct
+	void postConstruct() {
+
+		if (context.properties().useEndpointSlices()) {
```

Comment:
we need to make a distinction of when to use Endpoints vs EndpointSlices, because the latter might be not support on the k8s cluster. As such, we try to find out if it is supported or not and based on that return the proper `stateGenerator` instance

Reply:
No problem!